### PR TITLE
SOC-5515 : Optimize Social Query for activities, first phase

### PR DESCRIPTION
**Pull request by boubaker**
This PR will improve the activities queries and will avoid using a java query builder instead of NamedQuery. In fact, java query builder is CPU consumer and uses some blocker operations.
The first commits will be to improve the JPQL queries and a second commit will be added to delete the Query Builder and replace it by NamedQueries.

### CODE REVIEW

File: lib/src/main/java/org/exoplatform/social/addons/storage/RDBMSActivityStorageImpl.java [hunk 1]
```diff
@@ -866,39 +878,25 @@ public int getNumberOfOlderComments(ExoSocialActivity existingActivity, Long sin
...
-      if (existingActivity.getLikeIdentityIds() != null) updatedActivity.setLikerIds(new HashSet<>(Arrays.asList(existingActivity.getLikeIdentityIds())));
-      if (existingActivity.getMentionedIds() != null) updatedActivity.setMentionerIds(new HashSet<>(Arrays.asList(existingActivity.getMentionedIds())));
-      if (existingActivity.getPermaLink() != null) updatedActivity.setPermaLink(existingActivity.getPermaLink());
-      if (existingActivity.getTemplateParams() != null) updatedActivity.setTemplateParams(existingActivity.getTemplateParams());
-      updatedActivity.setHidden(existingActivity.isHidden());
-      updatedActivity.setComment(existingActivity.isComment());
-      updatedActivity.setLocked(existingActivity.isLocked());
-
-      activityDAO.update(updatedActivity);
+    ExoSocialActivity updatedActivity = getActivity(existingActivity.getId());
```
Comment (thomasdelhomenie) [inline lib/src/main/java/org/exoplatform/social/addons/storage/RDBMSActivityStorageImpl.java]: Is this change really related to the PR ? It seems to revert the fix https://github.com/exo-addons/social-rdbms/commit/22b61aeab8e6e8ca9754032728ffd7e5bc63767e

**Review by thomasdelhomenie — CHANGES_REQUESTED** [review]
Can you explain what you did in this PR please ?

**Comment by thomasdelhomenie** [comment]
I mean from a query point of view. I see you splitted the query to retrieve the connections in 2 queries (sender and receiver).

File: lib/src/main/java/org/exoplatform/social/addons/storage/RDBMSActivityStorageImpl.java (hunk 1 above)
Comment (boubaker) [inline lib/src/main/java/org/exoplatform/social/addons/storage/RDBMSActivityStorageImpl.java]: Nope, my bad.

**Comment by thomasdelhomenie** [comment]
Social RDBMS addon has been merged in Social project. Please move your PR.
Do we have a Jira issue for this PR ?

**Comment by boubaker** [comment]
Yep, I have added in title the JIRA issue: SOC-5515 (like branch name ;-) )

**Comment by boubaker** [comment]
moved to social project
https://github.com/exodev/social/pull/112

**Pull request closed without merging**